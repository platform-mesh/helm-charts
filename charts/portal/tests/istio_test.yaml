suite: deploy
chart:
  version: 1.0.0
  appVersion: 1.0.0
release:
  name: portal
tests:
  - it: matches the snapshot
    set:
      istio:
        enabled: false
    values:
      - ../test-values.yaml
    asserts:
      - matchSnapshot: {}
  - it: matches the snapshot with istio disabled
    values:
      - ../test-values.yaml
    set:
      istio:
        enabled: false
    asserts:
      - matchSnapshot: {}
