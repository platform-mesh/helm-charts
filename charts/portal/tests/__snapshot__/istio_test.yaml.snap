matches the snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: portal
      name: portal
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: portal
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
        type: RollingUpdate
      template:
        metadata:
          labels:
            app: portal
        spec:
          automountServiceAccountToken: true
          containers:
            - env:
                - name: HTTP_PROTOCOL
                  value: http
                - name: OIDC_CLIENT_ID_OPENMFP
                  value: openmfp
                - name: OIDC_CLIENT_SECRET_OPENMFP
                  valueFrom:
                    secretKeyRef:
                      key: attribute.client_secret
                      name: portal-client-secret-openmfp
                - name: TOKEN_URL_OPENMFP
                  value: http://openmfp-keycloak/keycloak/realms/openmfp/protocol/openid-connect/token
                - name: AUTH_SERVER_URL_OPENMFP
                  value: http://localhost:8000/keycloak/realms/openmfp/protocol/openid-connect/auth
                - name: CONTENT_CONFIGURATION_VALIDATOR_API_URL
                  value: http://openmfp-extension-manager-operator-server.openmfp-system.svc.cluster.local:8088/validate
                - name: BASE_DOMAINS_OPENMFP
                  value: localhost
                - name: OIDC_CLIENT_ID_PORTAL
                  value: "12345"
                - name: OIDC_CLIENT_SECRET_PORTAL
                  valueFrom:
                    secretKeyRef:
                      key: secret
                      name: portal-client-secret-portal
                - name: DISCOVERY_ENDPOINT_PORTAL
                  value: https://auth.example.com/.well-known/openid-configuration
                - name: CONTENT_CONFIGURATION_VALIDATOR_API_URL
                  value: null
                - name: BASE_DOMAINS_PORTAL
                  value: portal.example.com,portal2.example.com
                - name: IDP_NAMES
                  value: openmfp,portal
                - name: DEVELOPMENT_INSTANCE
                  value: "true"
                - name: FEATURE_TOGGLES
                  value: enableSessionAutoRefresh=true
                - name: PORT
                  value: "8080"
                - name: REGION
                  value: null
                - name: IMAGE_TAG
                  value: 1.0.0
                - name: IMAGE_NAME
                  value: ghcr.io/openmfp/portal
                - name: COOKIE_DOMAIN
                  value: localhost
                - name: FRONTEND_PORT
                  value: "8000"
                - name: ENVIRONMENT
                  value: local
                - name: VALID_WEBCOMPONENT_URLS
                  value: .?
                - name: TEST_TEST
                  value: test test
                - name: OPENMFP_PORTAL_CONTEXT_CRD_GATEWAY_API_URL
                  value: https://${org-subdomain}portal.dev.local:8443/api/kubernetes-graphql-gateway/root:orgs:${org-name}/graphql
                - name: OPENMFP_PORTAL_CONTEXT_IAM_SERVICE_API_URL
                  value: https://portal.dev.local:8443/iam/query
                - name: OPENMFP_PORTAL_CONTEXT_IAM_ENTITY_CONFIG
                  value: '{"account":{"contextProperty":"entityId"}}'
              image: ghcr.io/openmfp/portal:12345
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /rest/health
                  port: 9000
                periodSeconds: 10
              name: portal
              ports:
                - containerPort: 8080
                  name: http
                  protocol: TCP
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 9000
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /rest/health
                  port: 9000
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /rest/health
                  port: 9000
                periodSeconds: 10
              volumeMounts: null
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: portal
          terminationGracePeriodSeconds: 10
          volumes: null
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: contentconfigurations-reader
    rules:
      - apiGroups:
          - core.openmfp.io
        resources:
          - contentconfigurations
        verbs:
          - get
          - watch
          - list
  3: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: read-contentconfigurations-global
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: contentconfigurations-reader
    subjects:
      - kind: ServiceAccount
        name: portal
        namespace: NAMESPACE
  4: |
    apiVersion: v1
    imagePullSecrets:
      - name: github
    kind: ServiceAccount
    metadata:
      name: portal
  5: |
    apiVersion: v1
    kind: Service
    metadata:
      name: portal
    spec:
      ports:
        - name: http
          port: 8080
          protocol: TCP
      selector:
        app: portal
      type: ClusterIP
matches the snapshot with istio disabled:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: portal
      name: portal
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: portal
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
        type: RollingUpdate
      template:
        metadata:
          labels:
            app: portal
        spec:
          automountServiceAccountToken: true
          containers:
            - env:
                - name: HTTP_PROTOCOL
                  value: http
                - name: OIDC_CLIENT_ID_OPENMFP
                  value: openmfp
                - name: OIDC_CLIENT_SECRET_OPENMFP
                  valueFrom:
                    secretKeyRef:
                      key: attribute.client_secret
                      name: portal-client-secret-openmfp
                - name: TOKEN_URL_OPENMFP
                  value: http://openmfp-keycloak/keycloak/realms/openmfp/protocol/openid-connect/token
                - name: AUTH_SERVER_URL_OPENMFP
                  value: http://localhost:8000/keycloak/realms/openmfp/protocol/openid-connect/auth
                - name: CONTENT_CONFIGURATION_VALIDATOR_API_URL
                  value: http://openmfp-extension-manager-operator-server.openmfp-system.svc.cluster.local:8088/validate
                - name: BASE_DOMAINS_OPENMFP
                  value: localhost
                - name: OIDC_CLIENT_ID_PORTAL
                  value: "12345"
                - name: OIDC_CLIENT_SECRET_PORTAL
                  valueFrom:
                    secretKeyRef:
                      key: secret
                      name: portal-client-secret-portal
                - name: DISCOVERY_ENDPOINT_PORTAL
                  value: https://auth.example.com/.well-known/openid-configuration
                - name: CONTENT_CONFIGURATION_VALIDATOR_API_URL
                  value: null
                - name: BASE_DOMAINS_PORTAL
                  value: portal.example.com,portal2.example.com
                - name: IDP_NAMES
                  value: openmfp,portal
                - name: DEVELOPMENT_INSTANCE
                  value: "true"
                - name: FEATURE_TOGGLES
                  value: enableSessionAutoRefresh=true
                - name: PORT
                  value: "8080"
                - name: REGION
                  value: null
                - name: IMAGE_TAG
                  value: 1.0.0
                - name: IMAGE_NAME
                  value: ghcr.io/openmfp/portal
                - name: COOKIE_DOMAIN
                  value: localhost
                - name: FRONTEND_PORT
                  value: "8000"
                - name: ENVIRONMENT
                  value: local
                - name: VALID_WEBCOMPONENT_URLS
                  value: .?
                - name: TEST_TEST
                  value: test test
                - name: OPENMFP_PORTAL_CONTEXT_CRD_GATEWAY_API_URL
                  value: https://${org-subdomain}portal.dev.local:8443/api/kubernetes-graphql-gateway/root:orgs:${org-name}/graphql
                - name: OPENMFP_PORTAL_CONTEXT_IAM_SERVICE_API_URL
                  value: https://portal.dev.local:8443/iam/query
                - name: OPENMFP_PORTAL_CONTEXT_IAM_ENTITY_CONFIG
                  value: '{"account":{"contextProperty":"entityId"}}'
              image: ghcr.io/openmfp/portal:12345
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /rest/health
                  port: 9000
                periodSeconds: 10
              name: portal
              ports:
                - containerPort: 8080
                  name: http
                  protocol: TCP
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 9000
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /rest/health
                  port: 9000
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /rest/health
                  port: 9000
                periodSeconds: 10
              volumeMounts: null
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: portal
          terminationGracePeriodSeconds: 10
          volumes: null
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: contentconfigurations-reader
    rules:
      - apiGroups:
          - core.openmfp.io
        resources:
          - contentconfigurations
        verbs:
          - get
          - watch
          - list
  3: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: read-contentconfigurations-global
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: contentconfigurations-reader
    subjects:
      - kind: ServiceAccount
        name: portal
        namespace: NAMESPACE
  4: |
    apiVersion: v1
    imagePullSecrets:
      - name: github
    kind: ServiceAccount
    metadata:
      name: portal
  5: |
    apiVersion: v1
    kind: Service
    metadata:
      name: portal
    spec:
      ports:
        - name: http
          port: 8080
          protocol: TCP
      selector:
        app: portal
      type: ClusterIP
